components/Typography/src/UIFont+MaterialScalable.m

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "UIFont+MaterialScalable.h"
+
+#import <objc/runtime.h>
+
+#import "MDCTypography.h"
+#import "UIApplication+AppExtensions.h"
+
+static char MDCFontScaleObjectKey;
+
+@implementation UIFont (MaterialTypography)
```

Comment:
Looks like this can land before MDCFontScaler. Please consider breaking it into a separate PR with its own tests.  I have enough concerns about MDCFontScaler that it seems this will be delayed if it's not landed separately ahead of time.

Reply:
See general comment below, but I will argue that these two piece can and should land together.